File: test/dino.rb
require File.expand_path(File.join(File.dirname(__FILE__), "..", "xmpp/lib/xmpp"))
require File.expand_path(File.join(File.dirname(__FILE__), "..", "dino/lib/dino"))

require 'gio2'

DINO_SOURCE = File.expand_path(ENV['DINO_SOURCE_DIR'] ||= './git/dino/')
DINO_BUILD  = File.expand_path(ENV['DINO_BUILD_DIR']  ||= "#{ENV['DINO_SOURCE_DIR']}/build")

ENV['SEARCH_PATH'] ||= DINO_BUILD+"/plugins"

class App < Gio::Application
  type_register
  include Dino::Application 
  
  attr_accessor :db, :search_path_generator, :settings, :stream_interactor
  attr_writer   :plugin_registry
    
  def plugin_registry
    @plugin_registry ||= Dino::PluginsRegistry.new()
  end  

  install_property(GLib::Param::TypeInstance.new(
    "db", "db", "db",
    Dino::Database.gtype,
    224 | GLib::Param::READABLE | GLib::Param::WRITABLE
  ))
  
  install_property(GLib::Param::TypeInstance.new(
    "search-path-generator","Search Path Generator","sp sp",
    Dino::SearchPathGenerator,
    224 | GLib::Param::READABLE | GLib::Param::WRITABLE
  ))
  
  install_property(GLib::Param::Object.new(
    "settings","Settings","s s",
    Dino::EntitiesSettings,
    224 | GLib::Param::READABLE | GLib::Param::WRITABLE
  ))
  
  install_property(GLib::Param::TypeInstance.new(
    "plugin-registry","Plugin_registry","p p",
    Dino::PluginsRegistry,
    224 | GLib::Param::READABLE | GLib::Param::WRITABLE
  ))
  
  install_property(GLib::Param::TypeInstance.new(
    "stream-interactor","Stream IOnteracxtore","si p",
    Dino::StreamInteractor,
    224 | GLib::Param::READABLE | GLib::Param::WRITABLE
  ))    
  
  def initialize(app_id="im.dino.GIDino")
    super application_id: app_id, flags: :handles_open
    
    @search_path = ENV['SEARCH_PATH'] ||= DINO_BUILD
    self.search_path_generator = Dino::SearchPathGenerator.new(@search_path);
    
    Dino::Loader.reference_gobject plugin_registry
    
    init();
    
    signal_connect :activate do
      stream_interactor.interaction_modules
      quit
    end
    
    loader = Dino::PluginsLoader.new(self);
    loader.load_all();
  end
end

App.new.run

File: test/dino.py
import os
import sys
import gi

gi.require_version("Dino", "0.1")

from gi.repository import GLib, GObject, Pango, GdkPixbuf, Gtk, Gio, Qlite,Xmpp,Dino, DinoUI, Adw

def init_dino():
	Qlite.Database
	Xmpp.Jid
	Dino.Application
	

init_dino()

class Application(Gtk.Application, Dino.Application):
	__gtype_name__ = 'PyDinoApplication'
	
	plugin_registry       = GObject.Property(type=Dino.PluginsRegistry)
	search_path_generator = GObject.Property(type=Dino.SearchPathGenerator)
	db                    = GObject.Property(type=Dino.Database)
	stream_interactor     = GObject.Property(type=Dino.StreamInteractor)
	settings              = GObject.Property(type=Dino.EntitiesSettings)
	
	@search_path_generator.getter
	def search_path_generator(self):
		return self.sp
		
	@plugin_registry.getter
	def plugin_registry(self):
		return self.pr
	
	def __init__(self, n,f):
		super(Application,self).__init__(application_id=n, flags=f)
		
		self.sp=Dino.SearchPathGenerator.new(exec_path="/home/bidenboi/git/gi-dino/git/dino/build/plugins")

		self.pr = Dino.PluginsRegistry()

		self.init()
		
		self.connect("activate", self.on_activate)
		
		self.loader = Dino.PluginsLoader.new(self)
		self.loader.load_all()

	def on_activate(self, app):
		mp    = self.get_stream_interactor().get_module_by_type(Dino.MessageProcessor)
		calls = self.get_stream_interactor().get_module_by_type(Dino.Calls)
		
		self.quit()
		

def main(argv):
	dino = Application('dino.im.PyDino', Gio.ApplicationFlags.HANDLES_OPEN)
	dino.run(argv)


if __name__ == '__main__':
    SystemExit(main(sys.argv))
